app/uk/gov/hmrc/excisemovementcontrolsystemapi/services/WorkItemService.scala

```diff
@@ -33,10 +36,88 @@ class WorkItemService @Inject()
+    }
+
+  }
+
+  def rescheduleWorkItem(workItem: WorkItem[ExciseNumberWorkItem]): Future[WorkItem[ExciseNumberWorkItem]] = {
+
+    val ern = workItem.item.exciseNumber
+    val newFastPollRetriesLeft = Math.max(workItem.item.fastPollRetriesLeft - 1, 0)
+    val updatedItem = workItem.item.copy(fastPollRetriesLeft = newFastPollRetriesLeft)
+    val newAvailableAtTime = if (newFastPollRetriesLeft > 0) {
+      workItem.availableAt.plus(appConfig.workItemFastInterval.toJava)
+    } else {
+      workItem.availableAt.plus(appConfig.workItemSlowInterval.toJava)
+    }
+
+    val updatedWorkItem = workItem.copy(
+      item = updatedItem,
+      availableAt = newAvailableAtTime,
+      status = ToDo
+    )
+
+    workItemRepository.saveUpdatedWorkItem(updatedWorkItem)
+      .flatMap(_ => workItemRepository.getWorkItemForErn(ern))
+      .map { case seq if seq == Seq.empty => throw new RuntimeException("TODO") //TODO exception
+      case seq: Seq[WorkItem[ExciseNumberWorkItem]] => seq.head //Only one entry expected
+      }
+  }
+
+  def rescheduleWorkItemForceSlow(workItem: WorkItem[ExciseNumberWorkItem]): Future[WorkItem[ExciseNumberWorkItem]] = {
+    rescheduleWorkItem(workItem.copy(item = workItem.item.copy(fastPollRetriesLeft = 0)))
+  }
+
+  def markAs(id: ObjectId, status: ProcessingStatus, availableAt: Option[Instant] = None): Future[Boolean] =
+    workItemRepository.markAs(id, status, availableAt)
+
+  def pullOutstanding(failedBefore: Instant, availableBefore: Instant): Future[Option[WorkItem[ExciseNumberWorkItem]]] =
+    workItemRepository.pullOutstanding(failedBefore, availableBefore)
+
+  private def createWorkItem(ern: String): Future[WorkItem[ExciseNumberWorkItem]] = {
+
+    val nextAvailableAt = timestampService.timestamp().plus(appConfig.workItemFastInterval.toJava)
+
+    workItemRepository.pushNew(ExciseNumberWorkItem(ern, appConfig.fastIntervalRetryAttempts), nextAvailableAt)
+  }
+
+  private def updateWorkItemToRunOnFastIntervals(workItemToUpdate: WorkItem[ExciseNumberWorkItem]): Future[WorkItem[ExciseNumberWorkItem]] = {
+
+    val updatedItem = workItemToUpdate.item.copy(fastPollRetriesLeft = appConfig.fastIntervalRetryAttempts)
+    val ern = updatedItem.exciseNumber
+
+    val fastIntervalRunningTime = timestampService.timestamp().plus(appConfig.workItemFastInterval.toJava)
+
+    val nextAvailableAt = if (fastIntervalRunningTime.isBefore(workItemToUpdate.availableAt)) {
+      fastIntervalRunningTime
+    } else {
+      workItemToUpdate.availableAt
+    }
+
+    val updatedWorkItem = workItemToUpdate.copy(
+      item = updatedItem,
+      availableAt = nextAvailableAt,
+      receivedAt = timestampService.timestamp(), //lastSubmitted
+      status = ToDo,
+      failureCount = 0
+    )
+
+    //TODO what happens if Work Item is locked in the db because the job is running at the same time as we get a submission???
+
+    workItemRepository.saveUpdatedWorkItem(updatedWorkItem)
```

Comment:
would not be enough here just return the updateWorkItem if this is save? Do not understand why we are retrieving the same work item that we have saved (making an extra call).

Reply:
yeah should be. only thing out of date will be the updated time but shouldn't affect anything that being out of date.

Reply:
oh yes did not think about the update time.